llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -10558,11 +10562,15 @@ calculateSrcByte(const SDValue Op, uint64_t DestByte, uint64_t SrcIndex = 0,
 
     SrcIndex += BitShift / 8;
 
-    return calculateSrcByte(Op->getOperand(0), DestByte, SrcIndex, Depth + 1);
+    return calculateSrcByte(Op->getOperand(0), DestByte, SrcIndex, IsSigned,
+                            Depth + 1);
   }
 
   default: {
-    return ByteProvider<SDValue>::getSrc(Op, DestByte, SrcIndex);
+    if (auto L = dyn_cast<LoadSDNode>(Op))
```

Comment:
This seems concerning, this could miss other load times (e.g. does this catch atomic sextload?). I'd expect this to be with all load handling?

Reply:
I'm not convinced we should be combining atomics to begin with.

I've blacklisted atomics / memtrinsic until we have a better idea of how to handle the signedness of such. Primarily `isMemIntrinsic` will need to be handled as these should be supported ByteProviders.

Reply:
But you aren't combining atomics, you are just interpreting the result value

Reply:
Yes you are right -- 

Even so, I propose to blacklist for now.